### PR TITLE
feat(sdk) Add the snuba SDK to the tests

### PR DESCRIPTION
**Pull request by evanh**
Use the SDK in the tests. Duplicate a few tests as an integration smoke test.

### CODE REVIEW

File: tests/test_snql_sdk_api.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+        response = self.app.post("/sessions/snql", data=query.snuba())
+        data = json.loads(response.data)
+
+        assert response.status_code == 200
+        assert data["data"] == []
+
+    def test_join_query(self) -> None:
+        sp = Entity("spans", "s")
+        tr = Entity("transactions", "tr")
```
Comment (fpacifici) [inline tests/test_snql_sdk_api.py:108]: We are getting rid of the span entity soon. I would use events and groupedmessage.

**Review by fpacifici — COMMENTED** [review]
Do you plan to keep this dependency as permanent or is it just for the rollout ?

**Comment by evanh** [comment]
I had planned to leave this in. I think it's important to have some integration tests. While those will also exist in Sentry, having something closer to the source can help with debugging.

**Comment by fpacifici** [comment]
My problem with this dependency is that it goes in the wrong way. In order to run snuba we have to import the sdk, and it leaves the content of the sdk available to the snuba code base. This would prevent us from actually decoupling them. 
It would make things even harder if we wanted to export some symbols from snuba as a python package to be used by sentry or the sdk. This would cause a circular import.

I see the need for integration tests, mysql does a similar thing. It makes sense to me, but, in order to do that we should split requirements in requirements.txt and requirements-test.txt so that we import the sdk only in tests.

**Comment by evanh** [comment]
Makes sense. I'll split it out.

**Review by fpacifici — APPROVED** [review]

**Comment by fpacifici** [comment]
well, there are tests to fix.

**Pull request merged**